calour/dsfdr.py

```diff
@@ -229,79 +247,82 @@ def dsfdr(data, labels, method='meandiff', transform='rankdata', alpha=0.1, nump
+        u[crow,closepos]=t[crow]
 
     # calculate permutation p-vals
-    pvals = np.zeros([numbact])  # p-value for original test statistic t
-    pvals_u = np.zeros([numbact, numperm])  # pseudo p-values for permutated test statistic u
+    pvals=np.zeros([numbact]) # p-value for original test statistic t
+    pvals_u=np.zeros([numbact,numperm])  # pseudo p-values for permutated test statistic u 
     for crow in range(numbact):
-        allstat = np.hstack([t[crow], u[crow, :]])
-        allstat = 1 - (sp.stats.rankdata(allstat, method='min') / len(allstat))
-        pvals[crow] = allstat[0]
-        pvals_u[crow, :] = allstat[1:]
+        allstat=np.hstack([t[crow],u[crow,:]])
+        allstat=1-(sp.stats.rankdata(allstat,method='min')/len(allstat))
+        pvals[crow]=allstat[0]    
+        pvals_u[crow,:]=allstat[1:]
 
     # calculate FDR
-    if fdrmethod == 'dsfdr':
-        # sort the p-values for original test statistics from big to small
-        sortp = list(set(pvals))
-        sortp = np.sort(sortp)
-        sortp = sortp[::-1]
-
-        foundit = False
-        allfdr = []
-        allt = []
+    if fdrmethod=='dsfdr':
+        # sort the p-values for original test statistics from biggest to smallest
+        sortp=list(set(pvals))
+        sortp=np.sort(sortp)
+        sortp=sortp[::-1]
+
+        # find a data-dependent threshold for the p-value
+        foundit=False
+        allfdr=[]
+        allt=[]
         for cp in sortp:
-            realnum = np.sum(pvals <= cp)
-            fdr = (realnum + np.count_nonzero(pvals_u <= cp)) / (realnum * (numperm + 1))
+            realnum=np.sum(pvals<=cp)
+            fdr=(realnum+np.count_nonzero(pvals_u<=cp)) / (realnum*(numperm+1))
             allfdr.append(fdr)
             allt.append(cp)
-            if fdr <= alpha:
-                realcp = cp
-                foundit = True
-                break
+            if fdr<=alpha:
+                realcp=cp
+                foundit=True
+                break   
 
         if not foundit:
             # no good threshold was found
-            reject = np.repeat([False], numbact)
+            reject=np.repeat([False],numbact)
             return reject, tstat, pvals
 
         # fill the reject null hypothesis
-        reject = np.zeros(numbact, dtype=int)
-        reject = (pvals <= realcp)
-
-    elif fdrmethod == 'bhfdr':
+        reject=np.zeros(numbact,dtype=int)
+        reject=(pvals<=realcp)      
+        
+    elif fdrmethod=='bhfdr' or fdrmethod == 'filterBH':
         t_star = np.array([t, ] * numperm).transpose()
-        pvals = (np.sum(u >= t_star, axis=1) + 1) / (numperm + 1)
-        reject = statsmodels.sandbox.stats.multicomp.multipletests(pvals, alpha=alpha, method='fdr_bh')[0]
+        pvals=(np.sum(u>=t_star,axis=1)+1)/(numperm+1)
+        reject=statsmodels.sandbox.stats.multicomp.multipletests(pvals,alpha=alpha,method='fdr_bh')[0]
 
-    elif fdrmethod == 'byfdr':
+    elif fdrmethod=='byfdr':
```

Comment:
again in a lot of line here missing space before/after '=' '<=', '==' etc. and after ','
sorry, this pep8 is really annoying